docs/PWADevServer.md

```diff
@@ -0,0 +1,92 @@
+
+```js
+const path = require('path');
+const buildpack = require('@magento/pwa-buildpack');
+const PWADevServer = buildpack.Webpack.PWADevServer;
+
+module.exports = async env => {
+    const config {
+        /* webpack entry, output, rules, etc */
+
+        devServer: await PWADevServer.configure({
+            publicPath: '/pub/static/frontend/Vendor/theme/en_US/',
+            backendDomain: 'https://magento2.localdomain',
+            serviceWorkerFileName: 'sw.js',
+            paths: {
+                output: path.resolve(__dirname, 'web/js'),
+                assets: path.resolve(__dirname, 'web')
+            },
+            id: 'magento-venia'
+        })
+    };
+
+    config.output.publicPath = config.devServer.publicPath;
+
+    return config;
+}
+```
+
+- ⚠️ `PWADevServer.configure()` is async and returns a Promise, so a Webpack
+    config that uses it must use the [Exporting a Promise configuration type](https://webpack.js.org/configuration/configuration-types/#exporting-a-promise).
+    The newer `async/await` syntax looks cleaner than using Promises directly.
+- ⚠️ The emitted `devServer` object may have a custom `publicPath`. For best
+    ServiceWorker functionality, set `config.output.publicPath` to this value
+    once the `devServer` is created and before creating a ServiceWorker plugin.
+
+### Purpose
+
+The typical Webpack local development scenario uses [the devServer settings in
+`webpack.config.js`](https://webpack.js.org/configuration/dev-server/) to create
+a temporary local HTTP server for showing edits in real time.
+
+PWA development has a couple of particular needs:
+
+- The host must be _secure_ and _trusted_ to allow ServiceWorker operation
+- The host must be _unique_ to prevent ServiceWorker collisions
+
+urthermore, Magento PWAs are Magento 2 themes running on Magento 2 stores, so
+hey need to proxy backend requests to the backing store in a customized way.
+
+PWADevServer` handles all these needs:
+
+- Creates and caches a custom local hostname for the current theme
+- Adds the custom local hostname to `/etc/hosts`   🔐
+- Creates and caches an SSL certificate for the custom local hostname
+- Adds the certificate to the OS-level keychain so browsers trust it  🔐
+- Customizes the `webpack-dev-server` instance to:
+  - Proxy all asset requests not managed by Webpack to the Magento store
+  - Emulate the public path settings of the Magento store
+  - Automatically switch domain names in HTML attributes
+  - Debug or disable ServiceWorkers
+
+*The 🔐  in the above list indicates that you may be asked for a password at
+this step.*
+
+### API
+
+`PWADevServer` has only one method: `.configure(options)`. It returns a Promise
+for an object that can be assigned to the `devServer` property in Webpack
+configuration.
+
+#### `PWADevServer.configure(options: PWADevServerOptions): Promise<devServer>`
+
+#### `options`
+
+- `id: string`: **Required.** A unique ID for this project. Theme name is
```

Comment:
Would be useful to document what this id is used for (related to my other comment)